clouddrift/pairs.py

```diff
@@ -0,0 +1,508 @@
+        Second array of latitudes in degrees.
+    time1 : array_like, optional
+        First array of times.
+    time2 : array_like, optional
+        Second array of times.
+    space_tolerance : float, optional
+        Tolerance in meters for the spatial overlap. If the overlap is within
+        this tolerance, the pair is considered to be a chance pair. Default is
+        0, or no tolerance.
+    time_tolerance : float, optional
+        Tolerance for the temporal overlap. If the overlap is within this
+        tolerance, and the space_tolerance condition is satisfied, the pair is
+        considered a chance pair.. Default is 0, or no tolerance.
+
+    Returns
+    -------
+    indices1 : np.ndarray[int]
+        Indices of the first array that satisfy chance pair criteria.
+    indices2 : np.ndarray[int]
+        Indices of the second array that satisfy chance pair criteria.
+
+    Examples
+    --------
+    In the following example, we load the first two trajectories from the GLAD
+    dataset and find all longitudes, latitudes, and times that satisfy the
+    chance pair criteria of 6 km separation distance and no time separation:
+
+    >>> from clouddrift.datasets import glad
+    >>> from clouddrift.pairs import chance_pair
+    >>> from clouddrift.ragged import unpack
+    >>> ds = glad()
+    >>> lon1 = unpack(ds["longitude"], ds["rowsize"], rows=0).pop()
+    >>> lat1 = unpack(ds["latitude"], ds["rowsize"], rows=0).pop()
+    >>> time1 = unpack(ds["time"], ds["rowsize"], rows=0).pop()
+    >>> lon2 = unpack(ds["longitude"], ds["rowsize"], rows=1).pop()
+    >>> lat2 = unpack(ds["latitude"], ds["rowsize"], rows=1).pop()
+    >>> time2 = unpack(ds["time"], ds["rowsize"], rows=1).pop()
+    >>> chance_pair(lon1, lat1, lon2, lat2, time1, time2, 6000, np.timedelta64(0))
+    (array([-87.05255 , -87.04974 , -87.04612 , -87.04155 , -87.03718 ,
+            -87.033165], dtype=float32),
+     array([28.23184 , 28.227087, 28.222498, 28.217508, 28.2118  , 28.205938],
+           dtype=float32),
+     array([-87.111305, -87.10078 , -87.090675, -87.08157 , -87.07363 ,
+            -87.067375], dtype=float32),
+     array([28.217918, 28.20882 , 28.198603, 28.187078, 28.174644, 28.161713],
+           dtype=float32),
+     array(['2012-07-21T21:30:00.524160000', '2012-07-21T22:15:00.532800000',
+            '2012-07-21T23:00:00.541440000', '2012-07-21T23:45:00.550080000',
+            '2012-07-22T00:30:00.558720000', '2012-07-22T01:15:00.567360000'],
+           dtype='datetime64[ns]'),
+     array(['2012-07-21T21:30:00.524160000', '2012-07-21T22:15:00.532800000',
+            '2012-07-21T23:00:00.541440000', '2012-07-21T23:45:00.550080000',
+            '2012-07-22T00:30:00.558720000', '2012-07-22T01:15:00.567360000'],
+           dtype='datetime64[ns]'))
+    """
+    time_present = time1 is not None and time2 is not None
+
+    # If time is provided, subset the trajectories to the overlapping times.
+    if time_present:
```

Comment:
I guess this could be:
```suggestion
    if time_tolerance:
```
and you could remove the other one. Maybe [verifying](url) if we want to add a time tolerance, we need to have time1 and time2 parameters defined. As of right now, all of them are Optional so you would have to validate time_tolerance somehow.

Reply:
This won't work because `time_distance` of 0 or `np.timedelta64(0)` evaluate as False, and these are valid values under which we want to compute time collocation.